### PR TITLE
Support publishing of PR previews to the langium-previews repo

### PR DESCRIPTION
**Pull request by netomi**
As requested in https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4075.

Additional changes that needed to be made for previews to be correctly displayed:

- use relative links when generating the site
- update playground to load additional js files relatively instead of absolutely

### CODE REVIEW

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -1,34 +1,84 @@
...
+        with:
+          name: "site"
+          path: ./public
+      - uses: rossjrw/pr-preview-action@183082fd714654433c8e2f6daedbfb4f20f2a94a # v1.4.4
+        id: deployment
         with:
           source-dir: ./public
+          preview-branch: previews
+          umbrella-dir: pr-previews
+          deploy-repository: netomi/langium-previews
```
Comment (spoenemann) [inline .github/workflows/preview.yml:64]: Shouldn't this be `eclipse-langium/langium-previews`? (same in line 82 below)

Comment (netomi) [reply .github/workflows/preview.yml:64]: omg, copy and paste error

**Review by spoenemann — APPROVED** [review]
Let's try this 🚀

**Comment by msujew** [comment]
@netomi The preview deployment seems to fail. See [here](https://github.com/eclipse-langium/langium-website/actions/runs/7209042782/job/19639256100?pr=132). `rossjrw/pr-preview-action@183082fd714654433c8e2f6daedbfb4f20f2a94a` doesn't seem to be a valid commit.

**Comment by netomi** [comment]
Created a PR to unpin this action. The action checks out itself based on the tag and that does not seem to work with a commit hash.

**Pull request merged**